packages/p/pdfhummus/xmake.lua

```diff
@@ -36,14 +38,24 @@ package("pdfhummus")
             if has_config("libjpeg") then
                 add_requires("libjpeg")
             end
-            add_requires("freetype", "zlib", "libaesgm")
+            if not is_plat("windows","mingw") then
+                add_requires("libaesgm")
+            end
+            add_requires("freetype", "zlib")
             target("pdfwriter")
                 set_kind("$(kind)")
                 add_files("PDFWriter/*.cpp")
                 add_headerfiles("(PDFWriter/*.h)")
                 add_packages("freetype")
                 add_packages("libtiff", "libpng", "libjpeg")
-                add_packages("libaesgm", "zlib")
+                add_packages("zlib")
+                if is_plat("windows","mingw") then
+                    add_packages("zlib")
+                    add_files("LibAesgm/*.c")
+                    add_includedirs("LibAesgm")
```

Comment:
why not improve libaesgm package? it is from `pdfhummus/LibAesgm`.

Reply:
libaesgm is imported by pdfhummus at 2017, but its first release is earlier than 2009. It's hard to determine newer libaesgm can be compiled on windows.

Reply:
But now the libaesgm is exactly the same as the pdfhummus/LibAesgm, in addition, even if it does not compile on windows. We should also improve the package code of libaesgm. You can make libaesgm support Windows with add_patches